VenafiPS/Public/New-TppCertificate.ps1

```diff
@@ -267,6 +297,9 @@ function New-TppCertificate {
             $params.Body.Add('CustomFields', @($newCf))
         }
 
+        if ( $Device ) {
+            $params.Body.Add('Devices', $Device)
+        }
```

Comment:
I am concerned that this may be really difficult in practice to implement. Although you're right in that the 'Device' is simply an array of hashtables with both device and application parameters, it is really difficult to get this right so it will translate to the proper JSON when output.
My suggestion would be to have people add the device and applications as separate hashtables so they can a) add applications to an existing device and b) not have to figure out the proper way to structure the device hashtable with applications inside it.
Would be easier for the end user I think.

Reply:
how do we handle multiple devices/apps and know which app goes with which device if we split these out?

Reply:
going to require a 'DeviceName' key for the app hashtable to reference the device